corehq/apps/userreports/expressions/specs.py

```diff
@@ -263,3 +266,31 @@ def get_variables(self, item, context):
             for slug, variable_expression in self._context_variables.items()
         }
         return var_dict
+
+
+class FormsExpressionSpec(JsonObject):
+    type = TypeProperty('get_case_forms')
+    case_id_expression = DefaultProperty(required=True)
+
+    def configure(self, case_id_expression):
+        self._case_id_expression = case_id_expression
+
+    def __call__(self, item, context=None):
+        case_id = self._case_id_expression(item, context)
+
+        if not case_id:
+            return []
+
+        assert context.root_doc['domain']
+        return self._get_forms(case_id, context.root_doc['domain'])
+
+    def _get_forms(self, case_id, domain):
+        # todo, cahcing
```

Comment:
Should we cache forms by case_id so that other expressions in the same UCR spec can get cached versions. Is `quickcache` for around a minute good here?